components/backup-stream/src/metadata/client.rs

```diff
@@ -654,18 +663,35 @@ impl<Store: MetaStore> MetadataClient<Store> {
         self.meta_store.txn(txn).await
     }
 
+    pub async fn global_checkpoint_of(&self, task: &str) -> Result<Option<Checkpoint>> {
+        let cps = self.checkpoints_of(task).await?;
+        let mut min_checkpoint = None;
+        for cp in cps {
+            match cp.provider {
+                CheckpointProvider::Store(..) => {
+                    if min_checkpoint
+                        .as_ref()
+                        .map(|c: &Checkpoint| c.ts < cp.ts)
+                        .unwrap_or(true)
+                    {
+                        min_checkpoint = Some(cp);
+                    }
+                }
+                // The global checkpoint has higher priority than store checkpoint.
+                CheckpointProvider::Task | CheckpointProvider::Global => return Ok(Some(cp)),
+                CheckpointProvider::Region { .. } => continue,
```

Comment:
does this mean V2 checkpoint not works any more? 🤔

Reply:
If we detached there is a checkpoint V3, then we won't use V2.
BTW, `get_region_checkpoint` would try to get region checkpoint directly before calling this.